services/streammanager/kafka/kafkamanager.go

```diff
@@ -357,18 +366,6 @@ func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*Produ
 				return nil, fmt.Errorf("[Kafka] invalid SASL type: %w", err)
 			}
 		}
-
-		if destConfig.UseSSH {
```

Comment:
@saurav-malani I moved the SSH block out of the `if destConfig.SslEnabled {` condition like it was before.